ts/src/vector/constant/constVectors.spec.ts

```diff
@@ -0,0 +1,134 @@
+import { describe, it, expect } from "vitest";
+import { IntConstVector } from "./intConstVector";
+import { LongConstVector } from "./longConstVector";
+import BitVector from "../flat/bitVector";
+import { FlatSelectionVector } from "../filter/flatSelectionVector";
+
+function getSelectionIndices(selection: { selectionValues: () => number[]; limit: number }): number[] {
+    return selection.selectionValues().slice(0, selection.limit);
+}
+
+describe("IntConstVector", () => {
+    it("should return constant value", () => {
+        const vec = new IntConstVector("test", 42, 10);
+        expect(vec.size).toBe(10);
+        expect(vec.getValue(0)).toBe(42);
+        expect(vec.getValue(5)).toBe(42);
+        expect(vec.getValue(9)).toBe(42);
+    });
+
+    it("should filter matching constant", () => {
+        const vec = new IntConstVector("test", 42, 10);
+        const result = vec.filter(42);
+        expect(result.limit).toBe(10);
+        expect(getSelectionIndices(result)).toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    });
+
+    it("should return empty when filtering non-matching value", () => {
+        const vec = new IntConstVector("test", 42, 10);
+        const result = vec.filter(99);
+        expect(result.limit).toBe(0);
+    });
+
+    it("should match when value in array", () => {
+        const vec = new IntConstVector("test", 42, 5);
+        expect(vec.match([10, 42, 99]).limit).toBe(5);
+        expect(vec.match([10, 99]).limit).toBe(0);
+    });
+
+    it("should filter selected", () => {
+        const vec = new IntConstVector("test", 42, 10);
+        const selection = new FlatSelectionVector([2, 3, 4, 5]);
+        vec.filterSelected(42, selection);
+        expect(getSelectionIndices(selection)).toEqual([2, 3, 4, 5]);
+
+        const selection2 = new FlatSelectionVector([2, 3, 4]);
```

Comment:
Is this a different test? I would consider creating a different test for it if this tests something a bit different.
A good practice is a single `expect` in a unit test, I break this rule if I test different parts on the same object, but otherwise it's better to have a different test.